llvm/test/MC/ARM/pcrel-arm-ldr-imm8-relocs.s

```diff
@@ -0,0 +1,41 @@
+@ RUN: llvm-mc -filetype=obj -triple=armv7 %s -o %t
+@ RUN: llvm-readelf -r %t | FileCheck %s --check-prefix=ARM
+@ RUN: llvm-objdump -d --triple=armv7 %t | FileCheck %s --check-prefix=ARM_ADDEND
+
+@ ARM: R_ARM_LDRS_PC_G0
+@ ARM: foo1
+@ ARM: R_ARM_LDRS_PC_G0
+@ ARM: foo2
+@ ARM: R_ARM_LDRS_PC_G0
+@ ARM: foo3
+
+// The value format is decimal in these specific cases, but it's hex for other
+// ldr instructions. These checks are valid for both formats.
+
+@ ARM_ADDEND: r0, [pc, #-
+@ ARM_ADDEND 8]
+@ ARM_ADDEND: r0, [pc, #-
+@ ARM_ADDEND 8]
+@ ARM_ADDEND: r0, [pc, #-
+@ ARM_ADDEND 8]
```

Comment:
Generally we'd use a regex here to make the `0x` optional, something like `{{(0x)?}}`, at least that's what's done in other types of test file. That also means you don't get any weird stuff in the middle, `, #-stuffgoeshere8`, not that that's very likely here.

Reply:
Picking a file at random: llvm/test/Feature/load_plugin_error.ll

Reply:
Yes I don't know how you make it optional exactly, so I chose another route. The regex way is better ofc, I'll try to find out how to do it...

Reply:
As per usual there's a few ways and they are all indirectly documented. This is the closest I found for regex https://llvm.org/docs/TestingGuide.html

And not for this, but in general you'll see these https://llvm.org/docs/CommandGuide/lit.html#substitutions around.

I wouldn't commit them to memory though, just be aware they are out there and only a few "find in files" away if you need to know what they are.

Reply:
I was looking at lit not filecheck, here's the documentation: https://llvm.org/docs/CommandGuide/FileCheck.html#filecheck-regex-matching-syntax